api/serializers/substance.py

```diff
@@ -103,9 +117,40 @@ class Meta:
             + (
                 "cas_number",
                 "einec_number",
-                "max_quantity",
+                "max_quantity",  # une property du Model, set grâce à create() et update()
+                "max_quantities",
                 "nutritional_reference",
                 "unit",
             )
         )
         read_only = COMMON_READ_ONLY_FIELDS
+
+    @transaction.atomic
+    def create(self, validated_data):
+        # max_quantity doesn't exist in validated_data
+        max_quantity = self.initial_data.get("max_quantity")
```

Comment:
la création et MAJ ne fait rien pour la quantité max quand je le teste sur le front, est-ce que `initial_data` utilise les champs tjs en camelCase ?

En tout cas, AMA ici on peut garder le champ `max_quantity` au dessus, car on peut en servir des champs qui sont properties. Après dans le `create` et `update` on peut faire un `pop` sur le `validated_data` avant `super()...(validated_data)` et continuer faire la logique ici.

Reply:
La création, modification, delete sont testées dans https://github.com/betagouv/complements-alimentaires/pull/1812

@alemangui  validated_data ne contient pas le champ `max_quantity` car c'est une property de modèle, donc la validation supprime le champ. Il faut le chercher dans `initial_data`
C'est ce que j'ai voulu transmettre dans le commentaire "# max_quantity doesn't exist in validated_data"